conda/common/logic.py

```diff
@@ -29,26 +29,195 @@
 """
 from __future__ import absolute_import, division, print_function, unicode_literals
 
+from array import array
 from itertools import chain, combinations
-from logging import getLogger
+from logging import DEBUG, getLogger
 import pycosat
 
 from .compat import iteritems
+from .io import time_recorder
 
 log = getLogger(__name__)
 
 
+class ClauseList(object):
+    def __init__(self):
+        self._clause_list = []
+        self.append = self._clause_list.append
```

Comment:
These are some performance tweaks to avoid lookups. But this of course means, `self._clauses` should be modified in-place -- otherwise we have to update these methods when we were to reassign `self._clauses`.